bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerDeleteOp.java

```diff
@@ -94,4 +97,42 @@ public void safeOperationComplete(int rc, Void result) {
     public String toString() {
         return String.format("LedgerDeleteOp(%d)", ledgerId);
     }
+
+    public static class DeleteBuilderImpl  implements DeleteBuilder {
+
+        private long builderLedgerId = -1;
+        private final BookKeeper bk;
+
+        public DeleteBuilderImpl(BookKeeper bk) {
+            this.bk = bk;
+        }
+
+        @Override
+        public DeleteBuilder withLedgerId(long ledgerId) {
+            this.builderLedgerId = ledgerId;
+            return this;
+        }
+
+        @Override
+        public CompletableFuture<?> execute() {
+            CompletableFuture<Void> counter = new CompletableFuture<>();
+            delete(builderLedgerId, new SyncDeleteCallback(), counter);
+            return counter;
+        }
+
+        private void delete(long ledgerId, AsyncCallback.DeleteCallback cb, Object ctx) {
+            LedgerDeleteOp op = new LedgerDeleteOp(bk, ledgerId, cb, ctx);
```

Comment:
isn't it necessary to validate if the ledgerId is set or not?

Reply:
done